src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -425,8 +425,50 @@ rct_window* WindowSceneryOpen()
     if (window != nullptr)
         return window;
 
+    //count scenery groups to determine window width
+    int windowWidth = WINDOW_SCENERY_WIDTH;
+    auto maxTabs = 32;
+    int numTabs = 0;
+    for (ObjectEntryIndex scenerySetIndex = 0; scenerySetIndex < maxTabs; scenerySetIndex++)
+    {
+        const auto* sceneryGroupEntry = get_scenery_group_entry(scenerySetIndex);
+        if (sceneryGroupEntry != nullptr && scenery_group_is_invented(scenerySetIndex))
+        {
+            numTabs++;
+        }
+    }
+    switch (numTabs) {
```

Comment:
What i did was call max() with the original WINDOW_SCENERY_WIDTH const and 5 + numTabs * 31. I think since the window was always at least it should be done like this. Otherwise, if there is only one tab the window would be very narrow.